File: app/src/main/java/com/gpetuhov/samplemockito/dagger/modules/AppModule.java
package com.gpetuhov.samplemockito.dagger.modules;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import com.gpetuhov.samplemockito.SampleMockitoApp;
import com.gpetuhov.samplemockito.dagger.scopes.DataScope;
import com.gpetuhov.samplemockito.utils.PrefsUtils;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {
    @Provides
    @DataScope
    Context providesContext() {
        return SampleMockitoApp.application.getApplicationContext();
    }

    @Provides
    @DataScope
    SharedPreferences providesPrefs(Context context) {
        return PreferenceManager.getDefaultSharedPreferences(context);
    }

    @Provides
    @DataScope
    PrefsUtils providesPrefsUtils() {
        return new PrefsUtils();
    }
}

File: app/src/main/java/com/gpetuhov/samplemockito/dagger/modules/AppModule.kt
package com.gpetuhov.samplemockito.dagger.modules

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

import com.gpetuhov.samplemockito.SampleMockitoApp
import com.gpetuhov.samplemockito.dagger.scopes.DataScope
import com.gpetuhov.samplemockito.utils.PrefsUtils

import dagger.Module
import dagger.Provides

@Module
class AppModule {
    @Provides
    @DataScope
    fun providesContext(): Context {
        return SampleMockitoApp.application.applicationContext
    }

    @Provides
    @DataScope
    fun providesPrefs(context: Context): SharedPreferences {
        return PreferenceManager.getDefaultSharedPreferences(context)
    }

    @Provides
    @DataScope
    fun providesPrefsUtils(sharedPreferences: SharedPreferences): PrefsUtils {
        return PrefsUtils(sharedPreferences)
    }
}
